addin-fixed-weights/program/src/processor.rs

```diff
@@ -89,6 +104,55 @@ pub fn process_setup_voter_weight_record(
     Ok(())
 }
 
+/// Processes Set Delegate instruction
+pub fn process_set_vote_percentage_with_realm(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    vote_percentage: u16,
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+
+    let governance_info = next_account_info(account_info_iter)?; // 0
+    let realm_info = next_account_info(account_info_iter)?; // 1
+    let governing_token_mint_info = next_account_info(account_info_iter)?; // 2
+    let governing_token_owner_info = next_account_info(account_info_iter)?; // 3
+    let authority_info = next_account_info(account_info_iter)?; // 4
+    let token_owner_record_info = next_account_info(account_info_iter)?; // 5
+    let voter_weight_record_info = next_account_info(account_info_iter)?; // 6
+
+    let mut voter_weight_record = get_account_data::<VoterWeightRecord>(program_id, voter_weight_record_info)?;
+
+    if *realm_info.key != voter_weight_record.realm
+        || *governing_token_mint_info.key != voter_weight_record.governing_token_mint
+        || *governing_token_owner_info.key != voter_weight_record.governing_token_owner
+    {
+        return Err(VoterWeightAddinError::WrongVoterWeightRecordOwnership.into());
+    }
+
+    let owner_record_data = get_token_owner_record_data_for_seeds(
+        governance_info.key,
+        token_owner_record_info,
+        &get_token_owner_record_address_seeds(
+            realm_info.key,
+            governing_token_mint_info.key,
+            governing_token_owner_info.key,
+        ),
+    )?;
+    owner_record_data.assert_token_owner_or_delegate_is_signer(authority_info)?;
+
+    let voter_weight: u64 =
+        (voter_weight_record.voter_weight as u128)
+            .checked_mul(vote_percentage.into()).ok_or(VoterWeightAddinError::OverflowVoterWeight)?
+            .checked_div(10000).ok_or(VoterWeightAddinError::OverflowVoterWeight)?
+            .try_into().map_err(|_| VoterWeightAddinError::OverflowVoterWeight)?;
```

Comment:
You change `voter_weight` which is located in VoterWeightRecord. This will get the wrong result if the user changes `voter_percentage` several times. You should get the base value from VOTER_LIST!

P.S. what happens if the user specifies more than 10000 for the `vote_percentage`?